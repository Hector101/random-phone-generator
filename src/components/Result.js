import React from "react";
import saveAs from 'file-saver';


import '../styles//Result.scss';

class Result extends React.Component {
  _savePhoneNumbers = () => {
    const data = JSON.stringify(this.props.data, null, 2);

    saveAs(
      new Blob([data], {
        type: 'application/json;charset=utf-8'
      }),
      'database.json'
    );
  };

  render() {
    const { show, _closeModal, data, number } = this.props;

    const min = data.map(d => d.phoneNumber)
      .reduce((prev, next) => (next < prev ? next : prev ), data[0] && data[0].phoneNumber);

    const max = data.map(d => d.phoneNumber)
      .reduce((prev, next) => (next > prev ? next : prev ), '');

    return(
      <div className={`modal ${show && 'show'}`}>
        <div className="modal-content ba b--light-gray">
          <span
            className="close"
            onClick={_closeModal}
          >&times;</span>
          <div className="overview w-100 flex flex-column justify-start items-center">
            <span className="min w-50 tc pa2 mb1 bg-black-10"><span className="b">min number:</span> {min}</span>
            <span className="max w-50 tc pa2 mb1 bg-black-10"><span className="b">max number:</span> {max}</span>
            <span className="total w-50 tc pa2 mb2 bg-black-10"><span className="b">Amount Generated:</span> <span className="black">{number}</span></span>
            <button
              className="outline-0 w-30 mt2 pointer b pa2 ttu bw2 ba b--gray fw1 f6"
              onClick={this._savePhoneNumbers}
            >Save Results</button>
          </div>
        </div>
      </div> 
    )
  }
};

export default Result;
