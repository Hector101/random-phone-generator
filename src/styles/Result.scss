.modal {
  display: none;
  position: fixed;
  z-index: 1;
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: none;
  background-color: rgb(0,0,0);
  background-color: rgba(0,0,0,0.3);

  &.show {
    display: flex;
    flex: 1;
    justify-content: center;
    align-items: center;
  }

  .modal-content {
    width: 40%;
    height: 40%;
    background: #ffffff;

    .close {
      color: #c4bfbf;
      float: right;
      font-size: 2rem;
      font-weight: bold;
      margin-right: 0.5rem;
    }
    
    .close:hover,
    .close:focus {
      color: rgb(66, 65, 65);
      text-decoration: none;
      cursor: pointer;
    }
  }
}