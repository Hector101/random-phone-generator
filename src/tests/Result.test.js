import React from 'react';
import { shallow } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';

import Result from '../components/Result';

describe('Result', () => {
  const wrapper = shallow(<Result data={[{ id: 1, phoneNumber: '06372676237'}]}/>);

  it('components mounted successfully', () => {
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('should save data on _savePhoneNumbers call', () => {
    window.URL = {
      createObjectURL: () => {}
    };
    const spy = jest.spyOn(wrapper.instance(), '_savePhoneNumbers');

    wrapper.find('button').simulate('click');
  });
});
