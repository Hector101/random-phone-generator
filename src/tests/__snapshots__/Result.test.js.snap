// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result components mounted successfully 1`] = `
<div
  className="modal undefined"
>
  <div
    className="modal-content ba b--light-gray"
  >
    <span
      className="close"
    >
      Ã—
    </span>
    <div
      className="overview w-100 flex flex-column justify-start items-center"
    >
      <span
        className="min w-50 tc pa2 mb1 bg-black-10"
      >
        <span
          className="b"
        >
          min number:
        </span>
         
        06372676237
      </span>
      <span
        className="max w-50 tc pa2 mb1 bg-black-10"
      >
        <span
          className="b"
        >
          max number:
        </span>
         
        06372676237
      </span>
      <span
        className="total w-50 tc pa2 mb2 bg-black-10"
      >
        <span
          className="b"
        >
          Amount Generated:
        </span>
         
        <span
          className="black"
        />
      </span>
      <button
        className="outline-0 w-30 mt2 pointer b pa2 ttu bw2 ba b--gray fw1 f6"
        onClick={[Function]}
      >
        Save Results
      </button>
    </div>
  </div>
</div>
`;
